### DISCUSSION TITLE
i can't add new property

### DISCUSSION THREAD

**Discussion by Codeine999** [refinedev/refine#4020] [General]
hello i got some trouble i can't add a new property but i can add users, i already connected with cloudinary when i submit form nothing happened not error and in my console is show data that i printed ask chatgpt i try a lot but it not work this my code
`import Property from "../mongodb/models/property.js";
import User from "../mongodb/models/user.js";
import mongoose from "mongoose";
import * as dotenv from "dotenv";
import { v2 as cloudinary } from "cloudinary";

dotenv.config();

cloudinary.config({
    cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
    api_key: process.env.CLOUDINARY_API_KEY,
    api_secret: process.env.CLOUDINARY_API_SECRET,
});

const getAllProperties = async (req, res) => {};
const getPropertyDetail = async (req, res) => {};
const createProperty = async (req, res) => {
    try {
        const {
            title,
            description,
            propertyType,
            location,
            price,
            photo,
            email,
        } = req.body;

        const session = await mongoose.startSession();
        session.startTransaction();

        const user = await User.findOne({ email }).session(session);

        if (!user) throw new Error("User not found");

        const photoUrl = await cloudinary.uploader.upload(photo);

        const newProperty = await Property.create({
            title,
            description,
            propertyType,
            location,
            price,
            photo: photoUrl.url,
            creator: user._id,
        });

        user.allProperties.push(newProperty._id);
        await user.save({ session });

        await session.commitTransaction();

        res.status(200).json({ message: "Property created successfully" });
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};
const updateProperty = async (req, res) => {};
const deleteProperty = async (req, res) => {};

export {
    getAllProperties,
    getPropertyDetail,
    createProperty,
    updateProperty,
    deleteProperty,
}`
`const CreateProperty = () => {
  const navigate = useNavigate();
  const { data: user } = useGetIdentity<User>();
  const [propertyImage, setPropertyImage] = useState({ name: '', url: '' });
  const { refineCore: { onFinish, formLoading }, register, handleSubmit } = useForm();

  const handleImageChange = (file: File) => {
    const reader = (readFile: File) =>
      new Promise<string>((resolve, reject) => {
        const fileReader = new FileReader();
        fileReader.onload = () => resolve(fileReader.result as string);
        fileReader.readAsDataURL(readFile);
      });

    reader(file).then((result: string) =>
      setPropertyImage({ name: file?.name, url: result }),
    );
  };

  const onFinishHandler = async (data: FieldValues) => {
    if (!propertyImage.name) return alert("Please select an image");
  
    const response = await onFinish({
      ...data,
      photo: propertyImage.url,
      email: user?.email ?? '',
    });
  
    if (response && 'status' in response && response.status === 200) {
      // Success: Do something (e.g., navigate to another page)
      navigate('/path-to-some-page');
    } else if (response && 'message' in response) {
      // Error: Show an error message
      console.error('Failed to create property:', response.message);
    }
  };`

**Comment by BatuhanW** [comment]
Hello @Codeine999, I think this question isn't related to refine, you could be confusing repositories 😄 Feel free to re-open if it's relevant.
